Add missing NDT client values & Add tls & websocket columns

stephen-soltesz opened the pull request:
This PR addresses two issues:

 * client_kernel_version & client_version are parsed from .meta files addressing this issue from https://github.com/m-lab/etl/issues/168
 * "tls" and "websocket" are added as new columns with unit tests.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/m-lab/etl/172)
<!-- Reviewable:end -->

gfr10598 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>
Please rebase before submitting.

---

Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[parser/ndt_meta.go, line 80 at r1](https://reviewable.io:443/reviews/m-lab/etl/172#-KnplfFF1rjbELyCzkpl:-KnplfFF1rjbELyCzkpm:b8zq65i) ([raw file](https://github.com/m-lab/etl/blob/1848a7d8b952a6d1817713a3d2176791871a3467/parser/ndt_meta.go#L80)):*
> ```Go
> 		}
> 	}
> 	connSpec.SetBool("tls", mfd.Tls)
> ```

Should these be tri-valued, and handle the case where the meta file doesn't set them?  IIUC this code assumes they are set, so they will only be null if the meta file is missing.

---

*[parser/ndt_meta_test.go, line 47 at r1](https://reviewable.io:443/reviews/m-lab/etl/172#-Knpm7Blmkn22IEDDkeb:-Knpm7Blmkn22IEDDkec:bhm2d5y) ([raw file](https://github.com/m-lab/etl/blob/1848a7d8b952a6d1817713a3d2176791871a3467/parser/ndt_meta_test.go#L47)):*
> ```Go
> 	}
> 	if connSpec["websocket"] != true {
> 		t.Error("Incorrect websocket: got false; want")
> ```

want true?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/172#-:-KnpmKHGlKFymPETEm_0:bkvyjac)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
I fixed a typo for "websocket" -> "websockets" since that is the actual field name from ndt.

I believe I've preserved the tri-value state for the tls and websockets fields.

I updated `createMetaFileData` to additionally save the tls and websockets fields to `mfd.Fields`, since they were originally not saved only parsed, and for the tri-value state we need both.

Note: I have not updated the public tables with the additional columns.

---

Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[parser/ndt_meta.go, line 80 at r1](https://reviewable.io:443/reviews/m-lab/etl/172#-KnplfFF1rjbELyCzkpl:-KnppA9zT6vLneNrzC5A:b-9u0mup) ([raw file](https://github.com/m-lab/etl/blob/1848a7d8b952a6d1817713a3d2176791871a3467/parser/ndt_meta.go#L80)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

Should these be tri-valued, and handle the case where the meta file doesn't set them?  IIUC this code assumes they are set, so they will only be null if the meta file is missing.
</blockquote></details>

Good point. I've fixed this by making the call to `SetBool` conditional on whether the field is present and non-empty.

---

*[parser/ndt_meta_test.go, line 47 at r1](https://reviewable.io:443/reviews/m-lab/etl/172#-Knpm7Blmkn22IEDDkeb:-KnppGWCNFHQEa-1LibZ:bt2vr1b) ([raw file](https://github.com/m-lab/etl/blob/1848a7d8b952a6d1817713a3d2176791871a3467/parser/ndt_meta_test.go#L47)):*
<details><summary><i>Previously, gfr10598 (Gregory Russell) wrote…</i></summary><blockquote>

want true?
</blockquote></details>

Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/172#-:-KnpyTxf8r8spl7e8uHt:bos4byw)*
<!-- Sent from Reviewable.io -->

gfr10598 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/etl/172#-:-Knq0pP3Ip6NX78vLJF-:bnfp4nl)*
<!-- Sent from Reviewable.io -->
